<?xml version="1.0"?>

<launch>
    <node name="joy" pkg="joy" type="joy_node" output="screen"/>

    <node ns="shoulder" name="sin_gen" pkg="joy_convert" type="sin_generator.py" output="screen" >
        <param name="start_running" value="True" />
        <param name="amplitude" value="30.0" />
        <param name="outputOffset" value="60" />
        <param name="sin_period" value="1.5" />
        <remap from="waveform" to="/shoulder_servo_NC" />
        
    </node>

    <node ns="hip" name="sin_gen" pkg="joy_convert" type="sin_generator.py" output="screen" >
        <param name="amplitude" value="20" />
        <param name="outputOffset" value="60" />
        <param name="sin_period" value="4.0" />
        <param name="start_running" value="True" />
        <remap from="waveform" to="/hip_servo_NC" />
    </node>

    <node ns="shoulder" name="joy_convert" pkg="joy_convert" type="joy_convert.py" output="screen" >
        <param name="chNum" value="1" />
        <param name="minAngle" value="0" />
        <param name="maxAngle" value="120" />
        <param name="invert" value="True" />

        <remap from="joy" to="/joy" />
        <remap from="servo" to="/shoulder_servo" />
    </node>

    <node ns="hip" name="joy_convert" pkg="joy_convert" type="joy_convert.py" output="screen" >
        <param name="chNum" value="3" />
        <param name="minAngle" value="0" />
        <param name="maxAngle" value="120" />

i       <remap from="joy" to="/joy" />
        <remap from="servo" to="/hip_servo" />
    </node>

    <node name="arduino" pkg="rosserial_python" type="serial_node.py" output="screen">
        <param name="port" value="/dev/ttyACM0" />
    </node>

</launch>
